plugins/org.eclipse.elk.core/src/org/eclipse/elk/core/util/ElkUtil.java

```diff
@@ -535,7 +539,89 @@ private static KVectorChain determineJunctionPoints(final ElkEdge edge, final El
         return junctionPoints;
     }
 
+    /**
+     * Compute the bounding box of the labels of the port corresponding to the {@code port}. This bounding box
+     * coordinates are relative to the port (as the location of each port).
+     * 
+     * @param port
+     *            The port of the labels
+     * @return the bounds of the labels
+     */
+    public static ElkRectangle getLabelsBounds(final PortAdapter<?> port) {
+        ElkRectangle bounds = null;
+        for (LabelAdapter<?> label : port.getLabels()) {
+            ElkRectangle currentLabelBounds = new ElkRectangle(label.getPosition().x, label.getPosition().y,
+                    label.getSize().x, label.getSize().y);
+            if (bounds == null) {
+                // First label
+                bounds = currentLabelBounds;
+            } else {
+                bounds.union(currentLabelBounds);
+            }
+        }
+        if (bounds == null) {
+            // No label case, return an empty rectangle
+            bounds = new ElkRectangle();
+        }
+        return bounds;
+    }
 
+    /**
+     * Compute the part of the labels of the port that is inside the node. This method must be used for label with
+     * "fixed" port label placement ({@link PortLabelPlacement#isFixed(Set)}).
```

Comment:
It might be a good idea to throw an `IllegalArgumentException` or `IllegalStateException` if this assumption is violated.

Reply:
Yes, but I have not this information from a PortAdapter. I can change the signature of the method to pass PortContext (to have access to PortLabelPlacement.isFixed(parentNodeContext.portLabelsPlacement). OK with that?

Reply:
After some thought, adding a parameter just to verify that the caller respects the API seems strange to me.

Reply:
Ah, that's a good point; I didn't think of the fact that port adapters provide very limited information.

Well, not checking this here would keep the method behaving the same as its overloaded counterpart below, so that's probably a good idea anyway.

Reply:
If you're agree, I let you adapt with what is better according to you.